docs/secret_management.md

```diff
@@ -1,11 +1,128 @@
 # Secret Management
 
-Secret management is handled using the KSOPs plugin.
-
-Use [sops](https://github.com/mozilla/sops) to encrypt your secrets in vcs. Use the AICOE-SRE [public sops key](https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xefdb9afbd18936d9ab6b2eecbd2c73ff891fbc7e) to encrypt
+Secret management is handled using the KSOPs plugin. Use [sops](https://github.com/mozilla/sops) to encrypt your secrets in vcs. Use the AICOE-SRE [public sops key](https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xefdb9afbd18936d9ab6b2eecbd2c73ff891fbc7e) to encrypt
 your secrets so that ArgoCD may use KSOPs to decrypt them.
 
-## Updating Private key
+## Overview: KSOPs
+`KSOPS`, or kustomize-SOPS, is a [kustomize](https://github.com/kubernetes-sigs/kustomize/) plugin for SOPS encrypted resources. `KSOPS` can be used to decrypt any Kubernetes resource, but is most commonly used to decrypt encrypted Kubernetes Secrets and ConfigMaps. As a [kustomize](https://github.com/kubernetes-sigs/kustomize/) plugin, `KSOPS` allows you to manage, build, and apply encrypted manifests the same way you manage the rest of your Kubernetes manifests.
+
+## Requirements
+- [Go](https://github.com/golang/go)
+- [kustomize](https://github.com/kubernetes-sigs/kustomize/) built with Go (See [details below](#kustomize-go-plugin-caveats))
+- [SOPS](https://github.com/mozilla/sops)
+- gpg
+
+### 0. Verify Requirements
+Before continuing, verify your installation of [Go](https://github.com/golang/go), [SOPS](https://github.com/mozilla/sops), and `gpg`. Below are a few non-comprehensive commands to quickly check your installations:
+
+```bash
+# Verify Go is installed and your $GOPATH is set
+go env
```

Comment:
I initially ran into an issue due to having a version of go that was too old -- we may want to make a callout here for that.

Reply:
updated